packages/web/src/hooks/swap/data/use-swap-handler.tsx

```diff
@@ -104,6 +104,14 @@ function compareAmountFn(amountA: string | number | bigint, amountB: string | nu
 function handleAmount(changed: string, token: TokenModel | null) {
   let value = changed;
   const decimals = token?.decimals || 0;
+
+  // Check if input exceeds decimal places
+  const parts = changed.split(".");
+  if (parts.length > 1 && parts[1].length > decimals) {
+    // Signal invalid input
+    return { isValid: false, value: changed };
+  }
```

Comment:
This can be simplify, if using optional chain here.

```suggestion
  if (changed.split('.')[1]?.length > decimals) {
    return { isValid: false, value: changed };
  }
```

Reply:
```typescript
// Before
if (changed.split('.')[1]?.length > decimals) {
  return { isValid: false, value: changed };
}

// After
if (changed.includes('.') && changed.split('.')[1].length > decimals) {
  return { isValid: false, value: changed };
}
```

If check for decimal points first with `includes('.')` :
- More intuitive representation of decimal point presence
- Avoids unnecessary `split()` operations if no decimal point is present

Thank you for your suggestion:
- Optimize memory by removing unnecessary variable assignments
- Leaner, more intuitive code while maintaining the same behavior